packages/monitoring-system/data-processor/src/data-processors/github-data-processor.ts

```diff
@@ -84,58 +89,104 @@ export class GitHubProcessor extends DataProcessor {
 
   /**
    * List the GitHub repositories that have triggered
-   * bot executions in the past
+   * bot executions in the past and are not marked as private
    */
   private async listRepositories(): Promise<GitHubRepositoryDocument[]> {
     return this.firestore
       .collection(FirestoreCollection.GitHubRepository)
       .get()
       .then(repositoryCollection => {
-        const repositoryDocs = repositoryCollection.docs;
-        return repositoryDocs.map(
+        const firestoreDocs = repositoryCollection.docs;
+        const repositoryDocs = firestoreDocs.map(
           repoDoc => repoDoc.data() as GitHubRepositoryDocument
         );
+        const nonPrivateRepos = repositoryDocs.filter(
+          repoDoc => repoDoc.private === undefined || repoDoc.private === false
+        );
+        return nonPrivateRepos;
       });
   }
 
   /**
-   * Get all the publicly visible Events on the given repository
+   * Get recent 100 publicly visible Events on the given repository
    * @param repository repository for which to get events
+   * @returns A Promise with a list of GitHubEventDocuments. Promise
+   * will always resolve - if there is an error, it will resolve with
+   * an empty array and error will be logged.
+   *
+   * NOTE: We only fetch the first page (last 100) events given
```

Comment:
Will we be notified somehow if this happens? Or will we silently start missing events?

Reply:
No notification, but I think we'll notice if we start seeing bot executions without a mapped GitHub event on the dashboard.

The core issue here is that there's no definite 'ordering' to the events. In other words, if the latest event I have in Firestore has `timestamp = 1234`, I cannot guarantee that the next time I call the API I will have all the events with `timestamp < 1234` because the timestamp refers to the time of the event and not when the event was registered by the API. So an event with `timestamp=1134` may be registered at time `1344`. 

This leaves us with 2 possible solutions:

1. Make the change now to read all 300 events (this will triple the number of GitHub API calls)
2. Keep an eye on the dashboard and make the change later. At this point we'll probably also have to register the Data Processor as a GitHub app to increase the number of requests it can make.